frontend/src/components/globalSearch/GlobalSearchContent.tsx

```diff
@@ -175,7 +178,68 @@ export function GlobalSearchContent({
 
   // Resource search items
   const resources = useSearchResources();
+  const resourcesByKind = useMemo(
+    () => new Map(resources.map(resource => [resource.kind, resource])),
+    [resources]
+  );
   const loading = resources.filter(it => it.isLoading).map(it => it.kind);
+  const namespaceItems = (resourcesByKind.get(Namespace.kind)?.items ?? []) as Namespace[];
+  const namespaceOptions = useMemo(() => {
+    const knownNamespaces = new Set<string>();
+
+    namespaceItems.forEach(namespace => {
+      const name = namespace?.metadata?.name;
+      if (name) {
+        knownNamespaces.add(name);
+      }
+    });
+
+    selectedClusters.forEach(clusterName => {
+      const settings = loadClusterSettings(clusterName);
+      settings?.allowedNamespaces?.forEach(ns => {
+        if (ns) {
+          knownNamespaces.add(ns);
+        }
+      });
+    });
+
+    const options: SearchResult[] = [];
+
+    const addOption = (namespaceValue: string) => {
+      if (!namespaceValue) {
+        return;
+      }
+
+      options.push({
+        id: `set-namespace-${namespaceValue}`,
+        subLabel: t('translation|Namespace'),
+        label: t('translation|Set namespace: {{namespace}}', { namespace: namespaceValue }),
+        icon: (
+          <Suspense fallback={null}>
+            <LazyKubeIcon kind="Namespace" width="24px" height="24px" />
+          </Suspense>
+        ),
+        onClick: () => {
+          dispatch(setNamespaceFilter([namespaceValue]));
+        },
+      });
+    };
+
+    const trimmedQuery = query.trim();
+    if (trimmedQuery.length > 0) {
+      addOption(trimmedQuery);
+    }
+
+    Array.from(knownNamespaces)
+      .sort((a, b) => a.localeCompare(b))
+      .forEach(namespaceValue => {
+        if (namespaceValue !== trimmedQuery) {
```

Comment:
I'm not sure why this condition is needed. If I fully type a namespace, for example "default" then it won't show the option